### DISCUSSION TITLE
Setting up Grott Server

### DISCUSSION THREAD

**Discussion by westerguard** [johanmeijer/grott#582] [Q&A]
Hi all,

Currently, I'm running Grott proxy with the Home Assistant HACS add-on using a Shine Wifi-X.

Now I would like to run the Grott Server docker image. 
So I've installed it. It's running using an dedicated local IP.
I can access it on 'local ip:5782/inverter'

But the thing is, the Shine Wifi-X doesn't seem to recognize my Grott Server proxy on that local ip and port 5279.

The docker image should work out of the box, right? Or is there anything I forgot to configure?

If I look at the logs, I can see my pc connecting to the Grott Server, but I don't see anything else.
I've used both ledidobe/grottserver latest and beta. But no luck so far.

**Comment by dannybeeckman** [comment]
Probably the stick is connecting to the server.growatt.com address, you can rewrite that dns to your local grott IP.

**Comment by PedroKTFC** [comment]
This is my grott configuration and it works perfectly with grottserver. My grottserver is also running on my HA Rpi and that's its IP address below.

```
data_path: /config/grott
gmode: proxy
grott_mqtt: true
ginvtype: SPH
ginvtypemap: ""
retain: false
verbose: false
ha_plugin: true
mqtt: {}
ggrowattport: 5781
ggrowattip: 172.16.0.125
```

**Comment by westerguard** [comment]
@dannybeeckman well, I'm already using grott proxy in HA, So I know my inverter is ok, it's just not working with grott server.

@PedroKTFC thanks! I will try that. And the port used by the inverter/shine-wifi-lan is port 5781?

**Comment by johanmeijer** [comment]
Do you want to use the Grottserver container or grottserver python standalone program. 

The Grottserver docker container is  running both Grott and Grottserver and you should be able to run it replacing Grott without any changes except enabling the API port (if you want to use that one. 

docker run -d -p 5279:5279 -p 5782:5782 --name grott -e gmqttip="your.ip.address.number" ledidobe/grottserver:beta

I you run the grottserver standalone then you have to change your growatt server destination IP/port  of Grott  as described above. 

Be aware sometimes docker keeps the 5279 port assigned to the Grott container and you have to stop and remove that one (I am not sure under what conditions this happens).

**Comment by johanmeijer** [comment]
I did not realise you use the HA add on. Then you have to use the standalone grottserver.

**Comment by westerguard** [comment]
Thank you. I've found the culprit. 
I'm trying to set this up with Docker on Truenas with an external interface. So that I can put it in a separate IoT VLAN.

That's not working apparently. Don't know what it is exactly. But if I use it without the external interface (just use port mapping), the grott server shows that it's connected to my inverter.

With the external interface, I can access it on port 5782/inverter, but the inverter isn't showing up or anything.  

Anyway, thanks everyone!
Will have think about putting it in a VM, or let it run on the 'main' VLAN.
